fix(transformer/styled-components): should use source path to generate file hash when the path is absoulte path

Dunqing opened the pull request:
related: https://github.com/oxc-project/oxc/issues/11876#issuecomment-3058036852

overlookmotel commented:
### Merge activity

* **Jul 10, 4:24 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(transformer/styled-components): should use source path to generate file hash when the path is absoulte path (#12199)

related: https://github.com/oxc-project/oxc/issues/11876#issuecomment-3058036852
